x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -700,6 +737,38 @@ private LogicalPlan resolveFork(Fork fork, AnalyzerContext context) {
             return new Fork(fork.source(), fork.child(), newSubPlans);
         }
 
+        private LogicalPlan resolveRerank(Rerank rerank, List<Attribute> childrenOutput) {
+            List<Alias> newFields = new ArrayList<>();
+            boolean changed = false;
+
+            // First resolving fields used in expression
+            for (Alias field : rerank.rerankFields()) {
+                Alias result = (Alias) field.transformUp(UnresolvedAttribute.class, ua -> resolveAttribute(ua, childrenOutput));
+                newFields.add(result);
+                changed |= result != field;
+            }
+
+            if (changed) {
+                rerank = rerank.withRerankFields(newFields);
+            }
+
+            // Ensure the score attribute is resolved
+            if (rerank.scoreAttribute() instanceof UnresolvedAttribute ua) {
+                Attribute resolved = resolveAttribute(ua, childrenOutput);
+                if (resolved.resolved() == false) {
+                    resolved = ua.withUnresolvedMessage(format(null, "Missing required column [{}] for RERANK", MetadataAttribute.SCORE));
```

Comment:
I think I have a similar prob with RRF - we shouldn't require the `_score` column as an input of `Rerank`.
if the `_score` column does not exist, or it has a different type than double, `RERANK` should just add or overwrite the `_score` column.

Reply:
++, _score should not be needed.

Reply:
Did the necessary so the _score is not required anymore and it is added if it does not exists.